examples/jsm/nodes/shadernode/ShaderNode.js

```diff
@@ -449,6 +457,17 @@ export const tslFn = ( jsFunc ) => {
 
 addNodeClass( 'ShaderNode', ShaderNode );
 
+//
+
+export const setCurrentStack = stack => currentStack = stack;
+export const getCurrentStack = () => currentStack;
+
+export const If = ( ...params ) => currentStack.if( ...params );
+export const assign = ( ...params ) => currentStack.assign( ...params );
+export const append = ( ...params ) => currentStack.add( ...params );
+
+addNodeElement( 'append', append );
```

Comment:
How exactly `append` should work in chaining syntax, wouldn't it be too confusing?

Reply:
This appends a node to the current stack, we could interpret it as adding a line of code or a block of code to the current line. It would be used for void type nodes.